### PR TITLE
add easy to use chunked transfer extension header

### PR DESCRIPTION
**Pull request by fabge**
---
name: add easy to use header for transfer encoding chunked htmx extension
labels: chunked transfer
---

**Related Issue**
[Issue 1911 in htmx repo](https://github.com/bigskysoftware/htmx/issues/1911)

**Proposed Changes**
HTMX handles chunked transfer encoding of a response differently than in vanilla HTML. It waits for the complete response before displaying the response, instead of displaying already received chunks of content.

**Types of changes**
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist**
Go over all the following points, and put an `x` in all the boxes that apply:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I am aware that this is an nbdev project, and I have edited, cleaned, and synced the source notebooks instead of editing .py or .md files directly.

**Additional Information**
Any additional information, configuration or data that might be necessary to reproduce the issue.

### CODE REVIEW

File: fasthtml/core.py [hunk 1]
```diff
@@ -439,7 +440,7 @@ def __str__(self): return p
 class FastHTML(Starlette):
     def __init__(self, debug=False, routes=None, middleware=None, exception_handlers=None,
                  on_startup=None, on_shutdown=None, lifespan=None, hdrs=None, ftrs=None,
-                 before=None, after=None, ws_hdr=False,
+                 before=None, after=None, ws_hdr=False, ct_hdr=False,
```
Comment (pydanny) [inline fasthtml/core.py:443]: What does `ct_hdr` stand for? Chunked Transfer Header?

File: fasthtml/core.py [hunk 2]
```diff
@@ -455,8 +456,9 @@ def __init__(self, debug=False, routes=None, middleware=None, exception_handlers
         htmlkw = htmlkw or {}
         if default_hdrs:
             if surreal: hdrs = [surrsrc,scopesrc] + hdrs
-            if ws_hdr: hdrs = [htmxwsscr] + hdrs
-            if htmx: hdrs = [htmxscr] + hdrs
+            if ws_hdr: hdrs = [htmxwssrc] + hdrs
+            if ct_hdr: hdrs = [htmxctsrc] + hdrs
```
Comment (pydanny) [inline fasthtml/core.py:460]: ♻️ What does `ct_hdr` stand for?

File: nbs/api/00_core.ipynb [hunk 3]
```diff
@@ -1118,8 +1119,9 @@
     "        htmlkw = htmlkw or {}\n",
     "        if default_hdrs:\n",
     "            if surreal: hdrs = [surrsrc,scopesrc] + hdrs\n",
-    "            if ws_hdr: hdrs = [htmxwsscr] + hdrs\n",
-    "            if htmx: hdrs = [htmxscr] + hdrs\n",
+    "            if ws_hdr: hdrs = [htmxwssrc] + hdrs\n",
+    "            if ct_hdr: hdrs = [htmxctsrc] + hdrs\n",
```
Comment (pydanny) [inline nbs/api/00_core.ipynb:1123]: ♻️ What does `ct_hdr` stand for?

**Review by pydanny — APPROVED** [review]
The brevity and clarity of this PR is admirable. Offsetting the HTMX into a node package is also a bonus.

File: fasthtml/core.py (hunk 1 above)
Comment (fabge) [reply fasthtml/core.py:443]: exactly!

Comment (fabge) [reply fasthtml/core.py:443]: should we rename it? i wasn't able to come up with something clearer

**Comment by jph00** [comment]
Thank you, this looks good! Sorry I renamed the vars the other day, so there's a conflict now. Do you mind taking a look?

**Comment by jph00** [comment]
Many thanks for this. Would you mind taking a look at the conflict please @fabge ?

File: fasthtml/core.py (hunk 1 above)
Comment (pydanny) [reply fasthtml/core.py:443]: I like the name. Just wanted to be explicit for when I document it. 😄

**Pull request merged**